src/xcode/ENA/ENA/Source/Client/Security/CheckinEncryption/AESEncryption.swift

```diff
@@ -0,0 +1,66 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import CommonCrypto
+
+enum AESEncryptionError: Error {
+    case EncryptionFailed(Int)
+}
+
+struct AESEncryption {
+
+    private let encryptionKey: Data
+    private let initializationVector: Data
+
+    init(encryptionKey: Data, initializationVector: Data) {
+        self.encryptionKey = encryptionKey
+        self.initializationVector = initializationVector
+    }
+
+    func encrypt(data: Data) -> Result<Data, AESEncryptionError> {
+        return crypt(data: data, option: CCOperation(kCCEncrypt))
+    }
+
+    func decrypt(data: Data) -> Result<Data, AESEncryptionError> {
+        return crypt(data: data, option: CCOperation(kCCDecrypt))
+    }
+
+    private func crypt(data: Data, option: CCOperation) -> Result<Data, AESEncryptionError> {
+        let cryptedDataLength = data.count + kCCBlockSizeAES128
+        var cryptedData = Data(count: cryptedDataLength)
+        let keyLength = encryptionKey.count
+        let options = CCOptions(kCCOptionPKCS7Padding)
+        var bytesLength = Int(0)
+
+        let status = cryptedData.withUnsafeMutableBytes { cryptedBytes in
+            data.withUnsafeBytes { dataBytes in
+                initializationVector.withUnsafeBytes { ivBytes in
+                    encryptionKey.withUnsafeBytes { keyBytes in
+                        CCCrypt(
+                            option,
+                            CCAlgorithm(kCCAlgorithmAES),
+                            options,
+                            keyBytes.baseAddress,
+                            keyLength,
+                            ivBytes.baseAddress,
+                            dataBytes.baseAddress,
+                            data.count,
+                            cryptedBytes.baseAddress,
+                            cryptedDataLength,
+                            &bytesLength
+                        )
+                    }
+                }
+            }
+        }
+
+        guard UInt32(status) == UInt32(kCCSuccess) else {
```

Comment:
```suggestion
        guard Int32(status) == Int32(kCCSuccess) else {
```

I think the cast to `UIInt32` is not safe and can crash, as `status` is of type `CCCryptorStatus` which is a type alias for a signed Int32 and all values except `kCCSuccess` seem to be negative.

```
enum {
    kCCSuccess          = 0,
    kCCParamError       = -4300,
    kCCBufferTooSmall   = -4301,
    kCCMemoryFailure    = -4302,
    kCCAlignmentError   = -4303,
    kCCDecodeError      = -4304,
    kCCUnimplemented    = -4305,
    kCCOverflow         = -4306,
    kCCRNGFailure       = -4307,
    kCCUnspecifiedError = -4308,
    kCCCallSequenceError= -4309,
    kCCKeySizeError     = -4310,
    kCCInvalidKey       = -4311,
};
typedef int32_t CCStatus;
typedef int32_t CCCryptorStatus;
```

Reply:
Good point! I will change it to `status == kCCSuccess` in the next PR (see other comment).